compiler/erg_common/traits.rs

```diff
@@ -12,6 +14,128 @@ use crate::consts::{BUILD_DATE, GIT_HASH_SHORT, SEMVER};
 use crate::error::{ErrorDisplay, ErrorKind, Location, MultiErrorDisplay};
 use crate::{addr_eq, chomp, log, switch_unreachable};
 
+pub trait DequeStream<T>: Sized {
```

Comment:
I've chosen this name for now, but if something else is better, I'd appreciate suggestions.